scheduler/scheduler-server/src/main/java/org/ow2/proactive/scheduler/core/db/TaskDBUtils.java

```diff
@@ -0,0 +1,150 @@
+package org.ow2.proactive.scheduler.core.db;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import org.hibernate.Criteria;
+import org.hibernate.Session;
+import org.hibernate.criterion.Order;
+import org.hibernate.criterion.Property;
+import org.hibernate.criterion.Restrictions;
+import org.ow2.proactive.db.SortParameter;
+import org.ow2.proactive.scheduler.common.TaskSortParameter;
+import org.ow2.proactive.scheduler.common.task.TaskInfo;
+import org.ow2.proactive.scheduler.common.task.TaskState;
+import org.ow2.proactive.scheduler.core.db.TransactionHelper.SessionWork;
+
+
+public class TaskDBUtils {
+
+    public static SessionWork<List<TaskState>> taskStateSessionWork(final DBTaskDataParameters params) {
+
+        return new SessionWork<List<TaskState>>() {
+
+            @Override
+            @SuppressWarnings("unchecked")
+            public List<TaskState> executeWork(Session session) {
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) The Cyclomatic Complexity of this method "executeWork" is 16 which is greater than 10 authorized. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AMethodCyclomaticComplexity)